src/reader.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::file_pipe_log::LogFd;
+use crate::{Error, LogBatch, MessageExt, Result};
+
+use log::trace;
+
+#[derive(Debug)]
+pub struct LogBatchFileReader<'a> {
```

Comment:
Why read the full batch? You only need to read `(index, EntryIndex)` pair for recovery, that's why recovery can be faster with new log format.

Reply:
`LogBatchFileReader` does not read the full batch. Its read size is based on the max value of the entries_index size and a configurable `recovery_read_block_size` (see next pr), and caches if needed. This will reduce read times if log batches are small.